lib/template-loader.js

```diff
@@ -27,18 +28,18 @@ module.exports = function (content) {
   }
 
   if (!cons[opt.engine]) {
-    throw new Error(
+    emitError(new Error(
```

Comment:
We should just call `callback` here with the error as the first argument. Also need to return early, so `return callback(new Error('...'))`